### PR TITLE
A few more rubocop corrections and make a clean baseline

### LINKED ISSUE #796: Introduce a rubocop 'todo list'
Give Developers a clean baseline for new code and allow for existing offenses to be cleaned up over time. See rubocop [auto-generated config](https://docs.rubocop.org/rubocop/1.0/configuration.html#automatically-generated-configuration) docs.

### PR DESCRIPTION
**Pull request by smoline**
## Why
This brings the total number of rubocop offenses down from 1237 to 514. It also adds .rubocop_todo.yml so when rubocop is run it produces no offenses giving everyone a clean baseline. This PR should not be accepted before #798 (Better rubocop corrections) as this one builds on the work in that PR to reduce the overall number of todo items in .rubocop_todo.yml.

Resolves #796 Introduce a rubocop 'todo list'

### Pre-Merge Checklist

- [x] All new features have been described in the pull request
- [x] Security & accessibility have been considered
- [x] All outstanding questions and concerns have been resolved
- [x] Any next steps that seem like good ideas have been created as issues for future discussion & implementation
- [x] High quality tests have been added, or an explanation has been given why the features cannot be tested
- [x] New features have been documented, and the code is understandable and well commented
- [x] Entry added to CHANGELOG.md if appropriate

## What
All the cops handled in this pull request:

- [x] Layout/SpaceInsideBlockBraces
- [x] Layout/SpaceBeforeBlockBraces
- [x] Layout/SpaceAroundEqualsInParameterDefault
- [x] Layout/IndentationConsistency
- [x] Layout/FirstArgumentIndentation
- [x] Layout/ArgumentAlignment
- [x] Layout/MultilineBlockLayout
- [x] Layout/FirstHashElementIndentation
- [x] Layout/HashAlignment
- [x] Layout/BlockEndNewline
- [x] Layout/BlockAlignment
- [x] Layout/FirstArrayElementIndentation
- [x] Layout/ArrayAlignment
- [x] Layout/IndentationWidth
- [x] Layout/MultilineMethodCallIndentation
- [x] Layout/ArgumentAlignment
- [x] Layout/TrailingWhitespace
- [x] Layout/CommentIndentation
- [x] Layout/FirstHashElementIndentation
- [x] Layout/HashAlignment
- [x] Layout/ExtraSpacing
- [x] Style/HashSyntax
- [x] Layout/TrailingEmptyLines
- [x] Layout/EmptyLinesAroundBlockBody
- [x] Layout/ClosingParenthesisIndentation
- [x] Layout/SpaceBeforeComment
- [x] Layout/EmptyLinesAroundModuleBody
- [x] Layout/MultilineArrayBraceLayout
- [x] Layout/SpaceInsideArrayLiteralBrackets
- [x] Layout/MultilineOperationIndentation
- [x] Layout/EmptyLineAfterGuardClause
- [x] Layout/MultilineMethodCallBraceLayout
- [x] Layout/MultilineHashBraceLayout
- [x] Layout/EndAlignment
- [x] Layout/IndentationConsistency
- [x] Layout/ClosingParenthesisIndentation
- [x] Style/WordArray
- [x] Layout/ArrayAlignment
- [x] Layout/SpaceAfterComma
- [x] Layout/EmptyLinesAroundArguments
- [x] Layout/EmptyLinesAroundClassBody
- [x] Layout/SpaceInsideParens
- [x] Metrics/MethodLength
- [x] Style/PercentLiteralDelimiters
- [x] Layout/EmptyComment
- [x] Style/BlockDelimiters
- [x] Style/MethodDefParentheses
- [x] Layout/LineLength
- [x] Add .rubocop_todo.yml to give everyone a clean baseline

## How
I checked each file individually and made corrections that would not disturb how the code functioned. The easiest corrections are the only ones tackled in this PR. Each commit is by offense to make review easier.

### Testing

All tests were run to be sure they still passed. Rubocop is not normally tested, but was run to be sure the everyone would have a clean baseline.

## Outstanding Questions, Concerns and Other Notes

There are still 514 more rubocop offenses that need to be addressed carefully over time to improve the overall consistency and maintainability of the codebase.

### CODE REVIEW

**Comment by smoline** [comment]
@exbinary This one has been rebased on main and is ready for review.

File: .rubocop.yml [hunk 1]
```diff
@@ -12,6 +14,10 @@ AllCops:
 
 Layout/LineLength:
   Max: 120
+  Exclude:
+    - 'spec/models/html_sanitizer_spec.rb'
+    - 'config/initializers/devise.rb'
+    - 'app/models/importers/submission_response_importer.rb'
```
Comment (exbinary) [inline .rubocop.yml:20]: Wondering why these are listed here instead of in `.rubocop_todo.yml`?

**Review by exbinary — COMMENTED** [review]
@smoline , i finally got around to reviewing this; apologies for the long delay.

Looking great overall, thank you so much for all your work on this!!

Just a handful of comments below. If any of them are too much work to fix, i'm fine with merging as is and dealing with them later.

That said, `main` has probably changed a lot since you authored this branch, so i think some work is likely necessary to bring it up to date. LMK if you want help with this, i'd be happy to take the baton.

Happy new year!!

**Comment by exbinary** [comment]
@smoline , i want to give you an update on the RuboCop work you started.

I tried this week to merge `main` into this branch and bring the new code into alignment with the rules defined here. It turned out to be quite a lot of work!

In the meantime, https://github.com/testdouble/standard released v1.0 and i've had my eye on that as a potential alternative to a custom rubocop configuration.

Given all that, i'm thinking it doesn't make sense to try to merge this PR anymore 😞 .

I've opened #935 as an alternate approach. It does include one of your commits from #801. I'd love to hear your thoughts on this direction.

Of course, we are quite indebted to you for all your previous work getting our rubocop setup going! 🙏🏾 .

**Comment by exbinary** [comment]
The work here has been revived in #935. Closing with much appreciation for the groundwork here! 🙏🏾 🙌🏾

**Pull request closed without merging**